File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.java
import java.util.*;
import java.io.*;
public class Main {
    public static void main(String [] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer tk = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(tk.nextToken());

        int start=1;
        int end=1;
        int answer =1;
        long sum = start;
        while(end < n){
            if (sum==n){
                answer+=1;
                end+=1;
                sum+=end;
            }
            else if (sum < n) {
                end += 1;
                sum+=end;
            }
            else{
                sum-=start;
                start+=1;
            }
        }
        System.out.println(answer);

    }
}

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
# 수들의 합(실버5)
n = int(input())
start=1
end=1
answer=0
total=1
while end <= n:
    if total < n:
        end+=1
        total+=end
    elif total==n:
       answer+=1
       end+=1
       total+=end
    else:
        total-=start
        start+=1
print(answer)